File: security/src/main/java/top/bettercode/summer/security/GrantTypeUserDetailsService.java
package top.bettercode.summer.security;

import javax.servlet.http.HttpServletRequest;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

/**
 * @author Peter Wu
 */
public interface GrantTypeUserDetailsService extends UserDetailsService {
  UserDetails loadUserByGrantTypeAndRequest(String grantType, HttpServletRequest request)
      throws UsernameNotFoundException;

}

File: security/src/main/kotlin/top/bettercode/summer/security/userdetails/GrantTypeUserDetailsService.kt
package top.bettercode.summer.security.userdetails

import jakarta.servlet.http.HttpServletRequest
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService

/**
 * @author Peter Wu
 */
interface GrantTypeUserDetailsService : UserDetailsService {
    fun loadUserByGrantTypeAndRequest(grantType: String, request: HttpServletRequest): UserDetails
}
